### PR TITLE
New check-test-format tool

### PR DESCRIPTION
**Pull request by sergiocazzolato**
This tool is used to check spread tests formatting.

The change also includes a new test to validate the tool but also to check the format of the tests in the test suite

Then other projects can use this tool to validate the spread tasks format.

### CODE REVIEW

File: tools/check-test-format [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+    "execute",
+]
+MANDATORY_KEYS = ["summary", "execute"]
+
+
+def check_mandatory_keys(task_keys):
+    findings = []
+    for key in MANDATORY_KEYS:
+        if key not in task_keys:
+            findings.append("Key '{}' is mandatory".format(key))
```
Comment (Meulengracht) [inline tools/check-test-format:38]: Very very nitpicky: you can also use string interpolation by prepending 'f'
```append(f"Key '{key}' is mandatory")```

But you don't need to do this, just a FYI

**Review by Meulengracht — APPROVED** [review]
Looks really good, it's going to be awesome to have here!

**Comment by sergiocazzolato** [comment]
@Meulengracht perhaps we should have this is a dir called utils, because it is something to be used by tests.

What do you think?

File: tools/check-test-format (hunk 1 above)
Comment (sergiocazzolato) [reply tools/check-test-format:38]: ok, I'll make it in a following pr because I should make that change in all the file.

**Comment by Meulengracht** [comment]
> @Meulengracht perhaps we should have this is a dir called utils, because it is something to be used by tests.
> 
> What do you think?

Yea that makes sense, I think it would be nice to have test-specfic things seperated from the tools

**Pull request merged**